### ISSUE TITLE
Downloaded audiobooks don't save play progress correctly (android auto related?)

### ISSUE THREAD

**Issue by tehniemer** [advplyr/audiobookshelf-app#425]
### Steps to reproduce

1. Download audiobook
2. Play for a while
3. Resume much later (>~12 hours?)

### Expected behaviour
- Progress resumes from most recent stopping point.

### Actual behaviour
- Progress resumes from stopping point prior to the most recent one, but only sometimes.

### Environment data
I've noticed this behavior when using Android Auto to listen to audiobooks, it doesn't seem to happen 100% of the time and I'm not sure if it happens if the book is streamed from the server rather than downloaded and played. 

Audiobookshelf Version:

- [0.9.57-beta] Android App

#### Android Issue
Android version: 13

Device model: Pixel 6a

Stock or customized system: Stock

**Comment by higbee4** [comment]
This happens with downloaded books on iOS as well. I’m not sure about streaming since I always download. This happened frequently enough to me that I’ve stopped using ABS until it’s fixed.

**Comment by advplyr** [comment]
I think I may have an idea of why this happens now. It is specific to downloaded books and would assume that the progress is not actually updated on the server.
@mikehoyle You are checking that the app is showing the correct progress but can you also verify that the server has the correct progress?

**Comment by higbee4** [comment]
Just in case this info helps. On iOS, I have cellular data turned off for ABS. And I listen to books I downloaded over Wi-Fi. If I listen to a book when I’m not on Wi-Fi there is no way for the client to communicate my position back to the server until I’m back on Wi-Fi. So it would have to know to send that data the next time I connect with the server and not depend solely on a real-time connection.

**Comment by advplyr** [comment]
@higbee4 It should attempt to sync your offline progress whenever you connect back to the server

**Comment by higbee4** [comment]
That’s good to know. I think I have run into this problem with downloaded books without ever leaving a Wi-Fi connection. So the fact I may be offline for extended periods of time may not be a relevant factor to the issue. I just thought I’d mention it in case it was. Thanks.

**Comment by tehniemer** [comment]
I just ran into this again after updating to 0.9.61, 

![PXL_20230131_133839190.jpg](https://user-images.githubusercontent.com/11021632/216327800-41ee207a-4a91-4874-85f4-5ed299ee01fd.jpg)

![PXL_20230131_133832792.jpg](https://user-images.githubusercontent.com/11021632/216327801-7f2761cb-078c-40bd-8246-fd719ea57014.jpg)

**Comment by RobinD42** [comment]
I've also run into this with 0.9.61 over the past couple days. It's almost as if the app forgets that the downloaded book is associated with the book on the server. The progress is tracked locally for the downloaded file, but the book is not added to the Continue Listening shelf and the progress is not set on the server.

I tried to download some new books to try to further determine the specific details of the bug, (start playing the first time from the downloaded file. download, exit app then play, etc.) but the new downloads all work as expected. So maybe it was just with books I downloaded before the 0.9.61 app update?

I'll keep poking at it to see if I can spot any patterns.
